script.js

```diff
@@ -1,12 +1,304 @@
 //You can edit ALL of the code here
+/*fetches the the data from the Maze Api*/
+const rootElem = document.getElementById("root"); // parent of All my elements
+let allShows;
+let displayResult;
+
+fetch("https://api.tvmaze.com/shows")
+  .then((response) => response.json())
+  .then((shows) => {
+    allShows = shows;
+  });
+
+/* setup function will call all the functions that makes the different part of the page */
+/* 1 first part setup */
 function setup() {
-  const allEpisodes = getAllEpisodes();
-  makePageForEpisodes(allEpisodes);
+  displayResult = `The total is: 0 /${allShows.length}`; // display the number of episodes.
+  createHeader(displayResult); // Calls a function that creates the header section.
+  createDrdShows(); //Calls a drop list function.
+  displayEpisodeList();
+  makePageForShows(allShows);
+  createFooter(); // calls create footer function.
+}
+/*2 creates header*/
+
+/*Creates header function that displays the header element */
+function createHeader(displayResult) {
+  const headerElement = document.getElementsByTagName("header"); // Selects the header element based on the Id that it's have
+  const spanDisplayResult = document.createElement("span"); // Creates the span tag.
+  spanDisplayResult.id = "numberOfMatchingTitle"; // Assigns an id to the span tag.
+  spanDisplayResult.innerText = displayResult; // Assign the The total is: 0 /73 elements to span element.
+  headerElement[0].appendChild(spanDisplayResult); // Appended the span to the header element.
+
+  let theForm = document.createElement("form"); // Created the form element.
+  theForm.id = "search-box"; // Assigned it an id called search-box.
+  headerElement[0].appendChild(theForm); //Appended a form to header element.
+
+  let theInputText = document.createElement("input"); // Created the input Element.
+  theInputText.id = "userInput"; //Assigned an id to called userInput.
+  theInputText.type = "text"; //Set the Type to text.
+  theInputText.placeholder = "Search a film..."; //Assigned the placeHolder text to it.
+  theForm.appendChild(theInputText); //Appended the input to form.
+  theInputText.addEventListener("input", (event) => {
+    //eventListener that calls the search function.
+    search(theInputText.value.trim()); // Trimmed value of user input is passed to the search function.
+  });
+}
+function makePageForShows(showData) {
+  rootElem.textContent = ""; // Set the of the root to blank as my the root content is the parent of my elements
+
+  showData.forEach((e) => {
+    // Loops through the episode list
+    setShowHTML(e); // repeats set episode function to create the structure of the page.
+  });
+}
+function setShowHTML(show) {
+  let theContainer = document.createElement("div"); //Creates an div element assign it assign it to the container element.
+  theContainer.setAttribute("value", "");
+  theContainer.setAttribute("class", "xl-col-4 placeHolder"); // Assign the Class of "xl-col-4" from the grid CSS file.
```

Comment:
I prefer flex box in this kind of set many divs, it is interesting that we chose different ways to do it.